osu.Game.Tests/Visual/Navigation/TestSceneButtonSystemNavigation.cs

```diff
@@ -16,6 +17,42 @@ public partial class TestSceneButtonSystemNavigation : OsuGameTestScene
     {
         private ButtonSystem buttons => ((MainMenu)Game.ScreenStack.CurrentScreen).ChildrenOfType<ButtonSystem>().Single();
 
+        private GlobalActionContainer globalActionContainer => Game.ChildrenOfType<GlobalActionContainer>().First();
+
+        [Test]
```

Comment:
I'd definitely want to see test coverage of keyboard navigation getting interrupted by mouse, and then continuing from the last hovered button.

These are the kind of edge cases that require special attention for an implementation like this.